dxr/query.py

```diff
@@ -372,6 +413,25 @@ def filter_menu_items(plugins):
             if filter.description)
 
 
+def highlight_path(path, extents):
+    """Return list of path fragments containing <b> tags to describe highlighting based on given
+    extents.
+
+    :arg path: the path to highlight as a single string
+    :arg extents: iterable of unsorted, possibly overlapping (start, end) offset tuples
+
+    """
+    highlighted_content = highlight(path, extents)
+    # Split on slash, except the slash in </b>.
+    highlighted_fragments = []
+    for possibly_fragment in highlighted_content.split(sep):
```

Comment:
We actually shouldn't use os.path.sep here; it'll always be "/", no matter what. (In practice, it doesn't matter, since we support only UNIX, but stil…)